table/merging_iterator.cc

```diff
@@ -79,7 +79,6 @@ class MergingIterator : public InternalIterator {
     for (auto& child : children_) {
       child.DeleteIter(is_arena_mode_);
     }
-    status_.PermitUncheckedError();
```

Comment:
I hope we are careful about having `status_.PermitUncheckedError()` here. This status_ really needs to be checked if Valid() = false. This can unmask bug in DBIter or comapction iterator that fails to check the status code when an error happens.

Reply:
Sorry for the confusing. My point is that, I don't like that we would have to call status_.PermitUncheckedError(). We hope we still need to suppress it warning somewhere but in a better place.